curvefs/src/mds/heartbeat/metaserver_healthy_checker.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+ *  Copyright (c) 2021 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Created Date: 2021-09-16
+ * Author: chenwei
+ */
+
+#include <glog/logging.h>
+#include "curvefs/src/mds/heartbeat/metaserver_healthy_checker.h"
+#include "curvefs/src/mds/topology/topology_item.h"
+#include "curvefs/proto/topology.pb.h"
+
+using ::curvefs::mds::topology::MetaServer;
+using ::curvefs::mds::topology::TopoStatusCode;
+using ::curve::common::WriteLockGuard;
+using std::chrono::milliseconds;
+
+namespace curvefs {
+namespace mds {
+namespace heartbeat {
+void MetaserverHealthyChecker::CheckHeartBeatInterval() {
+    std::lock_guard<std::mutex> lockguard(hbinfoLock_);
```

Comment:
pay attention to this mutex, two functions will try to hold it, one is in here and another is `UpdateLastReceivedHeartbeatTime` which is called from the RPC thread. 

if task within the lock is time-consuming, it will block all RPC threads.

Reply:
done